uswds/uswds
# USWDS - Accessibility: Added button functionality to usa-button links

# Summary

**Added spacebar activation to links styled as buttons**. This allows users to trigger these links the same way they would expect to trigger buttons. `usa-card` buttons have been replaced with links styled as buttons.

This PR has two main goals based on @thisisdano's comments in #3690:
- Links that are styled as buttons should be triggered by the spacebar without using `role="button"` [^1]
- `usa-card` buttons should be replaced with links [^2]

[^1]: https://github.com/uswds/uswds/issues/3690#issuecomment-951091169
[^2]: https://github.com/uswds/uswds/issues/3690#issuecomment-951095032

## Breaking change

:warning: This is a breaking change.

`usa-card` buttons have been replaced with links styled as buttons.

Before: 

```html
<div class="usa-card__footer">
    <button type="button" class="usa-button">Visit Florida Keys</button>
</div>
```

After:
```html
<div class="usa-card__footer">
    <a href="#" class="usa-button">Visit Florida Keys</a>
</div>
```

## Related issue

Closes #3690 

## Related PRs
[Changelog PR →](https://github.com/uswds/uswds-site/pull/2085)

## Preview link
[usa-card w/ links styled as buttons](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-button-group-a11y/iframe.html?args=&id=components-card--default&viewMode=story)

Preview link:


## Problem statement

Links styled as buttons should be triggered as such

## Solution

Create a simple JS file to target all anchors with the `usa-button` class and allow them to be triggered with the spacebar

Replace buttons in `usa-card` with `usa-button` anchor tags

## Testing and review

1. Visit `button-group` storybook preview
2. Turn on screen reader
3. Tab to "Back" link
4. Verify the screen reader reads it as a link
5. Press spacebar key
6. Verify that the link is followed
7. Visit usa-card
8. Verify that all buttons are now links with `usa-button` class
9. Verify spacebar & enter button trigger the links
   - It may help to update the links `href` to verify the page updates

Before opening this PR, make sure you’ve done whichever of these applies to you:
- [x] Run `npm run prettier:js` to format any JS updates.

## Additional information

- Added `href="#"` to allow links to be targeted through tabbing. 
- Ran `npm run build:html` & `npm run build:storybook` for good measure.
